Source/RESTyard.ContractFirst/RESTyard.Generator/Hypermedia.xsd

```diff
@@ -85,25 +86,31 @@
 		<xs:attribute type="xs:string" name="name" use="required"/>
 		<xs:attribute type="hm:method" name="method" use="required"/>
 		<xs:attribute type="xs:string" name="title" use="optional"/>
-		<xs:attribute type="xs:string" name="parameterTypeName" use="optional"/>
-		<xs:attribute type="xs:string" name="resultDocument" use="optional"/>
+		<xs:attribute type="xs:IDREF" name="parameterTypeName" use="optional"/>
+		<xs:attribute type="xs:IDREF" name="resultDocument" use="optional"/>
 	</xs:complexType>
 
 	<!-- Parameter -->
 	<xs:complexType name="TransferParametersType">
 		<xs:sequence>
 			<xs:element name="Parameters" type="hm:ParameterType" maxOccurs="unbounded" minOccurs="0"/>
+			<xs:element name="ExternalParameters" type="hm:ExternalParameterType" maxOccurs="unbounded" minOccurs="0"/>
```

Comment:
I am not yet sure about the ExternalParameters.
On the one hand, in a legacy codebase one might have specially defined some parameter that cannot be generated as such, but may want to use it still (for now).
On the other hand, relying on the existence of records that are not defined by the XML means the contract is incomplete.

One use-case I can imagine is to add a check to the type="..." on <Property> to make sure the type exists. In that case, one would need the ability to define SystemTypes ("External Types") for the validation.
That would force users though to define a standard set of SystemTypes to be able to use int, double, string, etc.
It would also be more difficult to use IList<T> since each would have to be defined. But that creates problems in the TypeScript generator, so maybe we need a was to specify that a property is actually a collection of the named type.